app/models/potential_payment.rb

```diff
@@ -0,0 +1,28 @@
+class PotentialPayment < ApplicationRecord
+  has_paper_trail
+
+  REFERRAL = "referral"
+  CONTRIBUTION = "contribution"
+
+  belongs_to :payout_report
+  belongs_to :publisher
+  belongs_to :channel
+
+  validates :channel_id, presence: true, if: -> { kind == CONTRIBUTION}
+  validates :channel_id, presence: false, if: -> { kind == REFERRAL}
```

Comment:
Would you clarify your intention here? Did you want to make sure it's empty?

I'm going to test on my own side, but would you also verify that `presence: false` works? Looking here, I don't think `presence: false` works https://stackoverflow.com/questions/10070786/rails-3-validation-presence-false